met-web/src/components/userManagement/userDetails/AddToEngagement.tsx

```diff
@@ -1,22 +1,40 @@
 import React, { useContext, useEffect, useRef, useState } from 'react';
 import Modal from '@mui/material/Modal';
-import { Autocomplete, CircularProgress, Grid, Paper, Stack, TextField, useTheme } from '@mui/material';
+import {
+    Autocomplete,
+    CircularProgress,
+    FormControl,
+    FormControlLabel,
+    FormHelperText,
+    FormLabel,
+    Grid,
+    Paper,
+    Radio,
+    Stack,
+    TextField,
+    useTheme,
+} from '@mui/material';
 import { MetHeader3, MetLabel, MetSmallText, modalStyle, PrimaryButton, SecondaryButton } from 'components/common';
+import { USER_GROUP } from 'models/user';
 import { ActionContext } from './UserActionProvider';
 import { useForm, FormProvider, SubmitHandler, Controller } from 'react-hook-form';
 import { yupResolver } from '@hookform/resolvers/yup';
 import * as yup from 'yup';
 import { getEngagements } from 'services/engagementService';
+import { addUserToGroup } from 'services/userService/api';
 import { addTeamMemberToEngagement } from 'services/membershipService';
 import { When } from 'react-if';
 import { openNotification } from 'services/notificationService/notificationSlice';
 import { useAppDispatch, useAppSelector } from 'hooks';
 import { debounce } from 'lodash';
 import { Engagement } from 'models/engagement';
 import axios, { AxiosError } from 'axios';
+import { Palette } from 'styles/Theme';
+import ControlledRadioGroup from 'components/common/ControlledInputComponents/ControlledRadioGroup';
 
 const schema = yup
     .object({
+        group: yup.string().nullable(),
```

Comment:
try making group required if user does not have roles. You can move the schema inside the component and do something like this:
![Uploading image.png…]()

Reply:
updated, thanks